### PR TITLE
Bug 1834870: Use sync telemetry from application services

### PR DESCRIPTION
**Pull request by skhamis**
Should land after: https://github.com/mozilla/application-services/pull/5624
and after: https://github.com/mozilla/probe-scraper/pull/593

Moves sync manager telemetry from firefox-android into application-services:
  - This will put the metrics closer to the actual syncing of the records
  - also enables both android and iOS to use the same metrics across the board

The goal after that is to have the metrics be at the rust level  

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
3. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
4. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
5. Click on `View task in Taskcluster` in the new `DETAILS` section.
6. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1834870

### CODE REVIEW

**Comment by skhamis** [comment]
https://github.com/mozilla/application-services/pull/5588 has been approved but won't land until this has been also been approved (since it's a breaking change).

**Comment by skhamis** [comment]
The failure for the probe scraper will be fixed with: https://github.com/mozilla/probe-scraper/pull/593 which has the "greenlight" so this can be reviewed independently of that!

**Review by MatthewTighe — APPROVED** [review]
1 small nit/question, but this is otherwise LGTM!



image

**Comment by jonalmeida** [comment]
Drive-by: do we want to remove this telemetry over at the same time as it's being added to app services? Would there be a way to keep both to ensure there are no regressions before removing it from here since there is a "v2" in app services?

Have we done any testing to ensure we're still sending the right pings?

File: android-components/config/detekt-baseline.xml
Comment (skhamis) [reply android-components/config/detekt-baseline.xml:300]: ah yeah I think this got auto-updated, I'll remove!

**Comment by jonalmeida** [comment]
> reverting this would start the calls from the original ping again

Okay, that sounds fair enough and a smaller change to make in that scenario. Thanks!

**Pull request merged**
